cpp/src/Slice/Grammar.y

```diff
@@ -1774,10 +1784,14 @@ out_qualifier
 // ----------------------------------------------------------------------
 parameters
 // ----------------------------------------------------------------------
-: %empty
-{
-}
-| out_qualifier metadata optional_type_id
+: parameter_list
+| %empty
+;
+
+// ----------------------------------------------------------------------
+parameter_list
+// ----------------------------------------------------------------------
+: out_qualifier metadata optional_type_id
```

Comment:
Moving `%empty` out of `parameter_list` into an intermediate rule fixes the leading comma bug.
The old semantics was allowing a _parameter_ to be empty, as opposed to a _parameter list_ to be empty.
(Empty just means the empty string).

Ex: this was valid `void op(, int i);` because it was matching the empty string as the first parameter.